src/Symfony/Component/ObjectMapper/CachedObjectMapper.php

```diff
@@ -0,0 +1,350 @@
+    private ?\SplObjectStorage $objectMap = null;
+
+    public function __construct(
+        private readonly string $cacheDir,
+        ObjectMapperMetadataFactoryInterface $metadataFactory = new ReflectionObjectMapperMetadataFactory(),
+        ?PropertyAccessorInterface $propertyAccessor = null,
+        ?ContainerInterface $transformCallableLocator = null,
+        ?ContainerInterface $conditionCallableLocator = null,
+        ?ObjectMapperInterface $objectMapper = null,
+    ) {
+        $this->metadataFactory = $metadataFactory;
+        $this->propertyAccessor = $propertyAccessor;
+        $this->transformCallableLocator = $transformCallableLocator;
+        $this->conditionCallableLocator = $conditionCallableLocator;
+        $this->objectMapper = $objectMapper;
+    }
+
+    public function map(object $source, object|string|null $target = null): object
+    {
+        $objectMapInitialized = false;
+        if (null === $this->objectMap) {
+            $this->objectMap = new \SplObjectStorage();
+            $objectMapInitialized = true;
+        }
+
+        $sourceClass = $source::class;
+        $targetClass = $this->resolveTargetClass($source, $target);
+
+        if (!$targetClass) {
+            throw new MappingException(\sprintf('Mapping target not found for source "%s".', get_debug_type($source)));
+        }
+
+        if (\is_string($target) && !class_exists($target)) {
+            throw new MappingException(\sprintf('Mapping target class "%s" does not exist for source "%s".', $target, get_debug_type($source)));
+        }
+
+        $cacheKey = hash('xxh128', $sourceClass.'-to-'.$targetClass);
+
+        if (!isset($this->cachedMappings[$cacheKey])) {
+            $cacheFile = $this->cacheDir.'/'.$cacheKey.'.php';
+
+            if (!is_file($cacheFile)) {
+                $this->fs ??= new Filesystem();
+                $mappingData = $this->getMappingMetadata($source, $targetClass);
+                $code = $this->generateMappingCode($mappingData, $sourceClass, $targetClass);
+
+                $tmpFile = $this->fs->tempnam(\dirname($cacheFile), basename($cacheFile));
+                $this->fs->dumpFile($tmpFile, $code);
+                $this->fs->rename($tmpFile, $cacheFile, true);
+                $this->fs->chmod($cacheFile, 0o666 & ~umask());
+            }
+
+            $this->cachedMappings[$cacheKey] = require $cacheFile;
+        }
+
+        $mappingFunction = $this->cachedMappings[$cacheKey];
+        $mappingToObject = \is_object($target);
+        $targetObject = \is_object($target) ? $target : (new \ReflectionClass($targetClass))->newInstanceWithoutConstructor();
+
+        if ($source instanceof LazyObjectInterface) {
+            $source->initializeLazyObject();
+        } elseif (\PHP_VERSION_ID >= 80400) {
+            $refl = new \ReflectionClass($source);
+            if (!$refl->isUninitializedLazyObject($source)) {
+                $refl->initializeLazyObject($source);
+            }
+        }
```

Comment:
no need for the check:
```suggestion
            (new \ReflectionClass($source))->initializeLazyObject($source);
```